cfme/tests/automate/test_git_import.py

```diff
@@ -110,3 +122,80 @@ def test_automate_git_import_multiple_domains(request, appliance):
         request.addfinalizer(domain.delete_if_exists)
         assert not domain.exists
+
+
+@pytest.mark.tier(1)
+def test_refresh_git_current_user(imported_domain, new_user):
+    """
+    Polarion:
+        assignee: ghubale
+        initialEstimate: 1/8h
+        caseimportance: medium
+        caseposneg: positive
+        testtype: functional
+        startsin: 5.9
+        casecomponent: Automate
+        tags: automate
+        testSteps:
+            1. created non-super user 'user1' along with default 'admin' user.
+            2. Using admin user imported git repo.:
+               'https://github.com/ramrexx/CloudForms_Essentials.git' or any other repo.
+            3. Logged in with admin and refreshed domain- 'CloudForms_Essentials' or other domain.
+               Then checked all tasks.
+            4. Found user name 'admin' next to 'Refresh git repository'.
+            5. Then checked instances in that domain by logging in with user 'user1' and 'admin'.
+            6. Logged in with non-super user 'user1' and refreshed domain - 'CloudForms_Essentials'.
+               Then checked all tasks.
+            7. Found user name 'user1' next to 'Refresh git repository'.
+            8. Then checked instances in that domain by logging in with user 'user1' and 'admin'.
+        expectedResults:
+            1.
+            2.
+            3.
+            4.
+            5. It shows that
+               e.g. 'Automate Instance [Provisioning - Updated 2019-01-15 11:41:43 UTC by admin]'
+            6.
+            7.
+            8. It shows that
+               e.g. 'Automate Instance [Provisioning - Updated 2019-01-15 11:44:43 UTC by user1]'
+               Hence, correct user that calls refresh automation domain from git branch is shown.
+    Bugzilla:
+        1592428
+    """
+    tasks_collection = imported_domain.appliance.collections.tasks
+
+    # Refreshed imported domain by 'Admin' user
+    imported_domain.refresh(branch_or_tag='Branch', git_branch='origin/master')
+    view = imported_domain.create_view(DomainDetailsView)
+    view.flash.assert_message("Successfully refreshed!")
+    view = navigate_to(tasks_collection, 'AllTasks')
+
+    # Collecting list of all tasks performed by users
+    all_tasks = view.tabs.alltasks.table.read()
+    expected_task = None
+    for task in all_tasks:
+        if task['Task Name'] == 'Refresh git repository':
+            expected_task = task
+            break
+    if expected_task:
+        assert expected_task['User'] == 'admin'
+    else:
+        raise NameError("Task not found")
+
+    with new_user:
+        # Refreshed imported domain by non-super user
+        imported_domain.refresh(branch_or_tag='Branch', git_branch='origin/master')
+        view = navigate_to(tasks_collection, 'AllTasks')
+
+        # Collecting list of all tasks performed by users
+        all_tasks = view.tabs.alltasks.table.read()
+        expected_task = None
+        for task in all_tasks:
```

Comment:
This and the similar case above can be written bit better using for/else
https://book.pythontips.com/en/latest/for_-_else.html#else-clause
but this is optional

Reply:
Since I'm requesting some other changes, consider changing this implementation @ganeshhubale. It will clean this up and make it more readable.  No need to initialize the `expected_task` or explicitly check it.

Reply:
@mshriver @jaryn 
- I have updated with ```for/else```. 
- Please take a look:
```
 all_tasks = view.tabs.alltasks.table.read()
    for task in all_tasks:
        if task['Task Name'] == 'Refresh git repository':
            assert task['User'] == 'admin'
            break
    else:
        raise NameError("Task not found")

```